native_client/README.md

```diff
@@ -1,105 +1,137 @@
 
+```
+PREFIX=/usr/local sudo make install
+```
 
-### Building your own Binaries
+It is assumed that `$PREFIX/lib` is a valid library path, otherwise you may need to alter your environment.
 
+### Python bindings
 
-Firstly, you should create a symbolic link in your TensorFlow checkout to the DeepSpeech `native_client` directory. If your DeepSpeech and TensorFlow checkouts are side by side in the same directory, do:
+Included are a set of generated Python bindings. After following the above build and installation instructions, these can be installed by executing the following commands (or equivalent on your system):
 
 ```
-cd tensorflow
-ln -s ../DeepSpeech/native_client ./
+cd native_client/python
+make bindings
+pip install dist/deepspeech*
 ```
 
-Next, you will need to prepare your environment to configure and build TensorFlow. Clone from `https://github.com/mozilla/tensorflow`, and then (preferably) checkout the version of `tensorflow` which is currently supported by DeepSpeech (see requirements.txt), and use the `bazel` version recommended by TensorFlow for that version. Follow the [instructions](https://www.tensorflow.org/install/install_sources) on the TensorFlow site for your platform, up to the end of ["Configure the Build"](https://www.tensorflow.org/install/source#configure_the_build).
+The API mirrors the C++ API and is demonstrated in [client.py](python/client.py). Refer to [deepspeech.h](deepspeech.h) for documentation.
+
+### Node.JS bindings
 
-After that, you can build the main DeepSpeech library, `libdeepspeech.so`, as well as the `generate_trie` binary using the following command:
+After following the above build and installation instructions, the Node.JS bindings can be built:
 
 ```
-bazel build --config=monolithic -c opt --copt=-O3 --copt="-D_GLIBCXX_USE_CXX11_ABI=0" --copt=-fvisibility=hidden //native_client:libdeepspeech.so //native_client:generate_trie
+cd native_client/javascript
+make package
+make npm-pack
 ```
 
-If your build target requires extra flags, add them. For example `--config=cuda` if you want a CUDA build. Note that the generated binaries will show up under `bazel-bin/native_client/` (e.g., including `generate_trie` in case the `//native_client:generate_trie` option was present).
+This will create the package `deepspeech-VERSION.tgz` in `native_client/javascript`.
+
+### Building the CTC decoder package
 
-Finally, you can change to the `native_client` directory and use the `Makefile`. By default, the `Makefile` will assume there is a TensorFlow checkout in a directory above the DeepSpeech checkout. If that is not the case, set the environment variable `TFDIR` to point to the right directory.
+To build the `ds_ctcdecoder` package, you'll need the general requirements listed above (in particular SWIG).
 
 ```
-cd ../DeepSpeech/native_client
-make deepspeech
+cd native_client/ctcdecode
+make bindings NUM_PROCESSES=`nproc`
```

Comment:
nproc is part of coreutils and will not be available by default on macOS hosts. Please keep the old docs with a number and instructions on what it means.

Reply:
fixed: 7ab0962b3d31c0d9f89238da2962c2edce998e7b